core/src/main/groovy/noe/ews/workspace/WorkspaceEws.groovy

```diff
@@ -28,9 +28,7 @@ class WorkspaceEws extends WorkspaceAbstract {
   String ewsVersion /// like 2.0.0-ER3
 
   WorkspaceEws(boolean installHttpd = true) {
-    if(installHttpd) {
-      workspaceHttpd = new WorkspaceHttpd()
-    }
+    workspaceHttpd = installHttpd ? new WorkspaceHttpd() : null
```

Comment:
What is the point of this change? workspaceHttpd remains uninitialized, what set it to null explicitly?

Reply:
This is strange. So you tried it and this change works for you..?

Reply:
@mmadzin can you email me links to failing jobs with description what are you trying to solve please?

Reply:
Matus wrote: The stack trace can be found below. When I used original code which is in MR, it works. 
`
java.lang.RuntimeException: Server with id httpd does not exist.
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:83)
        at org.codehaus.groovy.reflection.CachedConstructor.doConstructorInvoke(CachedConstructor.java:77)
        at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrap.callConstructor(ConstructorSite.java:84)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:60)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:235)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:247)
        at noe.server.ServerController.getServerById(ServerController.groovy:269)
        at noe.server.ServerController$getServerById$3.call(Unknown Source)
        at noe.workspace.WorkspaceHttpd.installHttpd(WorkspaceHttpd.groovy:71)
        at noe.workspace.WorkspaceHttpd.prepare(WorkspaceHttpd.groovy:79)
        at noe.workspace.WorkspaceHttpd.prepare(WorkspaceHttpd.groovy)
        at noe.workspace.IWorkspace$prepare.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callSafe(AbstractCallSite.java:80)
        at noe.ews.workspace.WorkspaceEws.prepare(WorkspaceEws.groovy:47)
        at noe.workspace.IWorkspace$prepare.call(Unknown Source)
        at noe.common.tests.TestSuiteAbstract.prepareWorkspace(TestSuiteAbstract.groovy:32)
        at noe.common.tests.TestSuiteAbstract$prepareWorkspace.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:117)
        at noe.tomcat.TestSuiteTomcat.prepareWorkspace(TestSuiteTomcat.groovy:53)`

Reply:
I skimmed the stacktrace, I don't see any relation at the moment. 

However, isn't 2nd change enough to solve the problem?

Reply:
@jstefl I reverted this part of the code to the original version and I got the stack trace. When the code is as I suggest in MR, it works.

Reply:
Could you please share links to jenkins?

Reply:
The question is, why server id for httpd is not set. According stack trace, httpd should be installed.

What did you found in logs? Is httpd installed and server id is not set? Or is server id lost somehow latter in process?

Reply:
@jstefl No I can't share Jenkins links. 

I don't care much about httpd installation or whether id was lost, because in the first place I don't want to install httpd, hence the variable should contain null and that's the suggested change - install the httpd server or set null to variable.

Reply:
In that case, you shouldn't install it. According stack trace you do it. Fix here is workaround for for non desired behavior.

Reply:
It's not a workaround because "installHttpd" variable contains boolean whether httpd should be installed a.k.a workspaceHttp is created.

Reply:
We found out the root of the issue. Variable workspaceHttpd is marked as a static which means that it shares among different instances. Hence if some other instance was created with httpd before then workspaceHttpd is not null.

Reply:
OK... I moved the null assignment to destroy method.